components/availability/src/Availability.svelte

```diff
@@ -413,31 +407,52 @@
               end_time: endTime,
               available_calendars: [],
             };
+
+            // Adjust calendar.timeslots for buffers
             const timeslots =
               calendar.availability === AvailabilityStatus.BUSY
-                ? calendar.timeslots.map((t) => ({
+                ? calendar.timeslots.map((t: TimeSlot) => ({
                     start_time: timeMinute.offset(t.start_time, -event_buffer),
                     end_time: timeMinute.offset(t.end_time, event_buffer),
                     available_calendars: t.available_calendars,
                   }))
-                : calendar.timeslots;
-            const slotAvailability = overlap(timeslots, slot);
+                : calendar.timeslots.map((t: TimeSlot) => ({
```

Comment:
I think this would have been a great idea up until the latest change, which added a nested ternary to avoid a buffer at the start of the first timeslot of the day / end of the last timeslot of the day.